src/Ocelot.Provider.Consul/ConsulProviderFactory.cs

```diff
@@ -1,41 +1,41 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
 using Ocelot.Configuration;
 using Ocelot.Logging;
 using Ocelot.Provider.Consul.Interfaces;
 using Ocelot.ServiceDiscovery.Providers;
 
 namespace Ocelot.Provider.Consul;
 
-public static class ConsulProviderFactory
+/// <summary>
+/// TODO It must be refactored converting to real factory-class and add to DI.
+/// </summary>
+/// <remarks>
+/// Must inherit from <see cref="IServiceDiscoveryProviderFactory"/> interface.
+/// Also the <see cref="ServiceDiscoveryFinderDelegate"/> must be removed from the design.
+/// </remarks>
+public static class ConsulProviderFactory // TODO : IServiceDiscoveryProviderFactory
 {
-    /// <summary>
-    /// String constant used for provider type definition.
-    /// </summary>
+    /// <summary>String constant used for provider type definition.</summary>
     public const string PollConsul = nameof(Provider.Consul.PollConsul);
 
-    private static readonly List<PollConsul> ServiceDiscoveryProviders = new();
-    private static readonly object LockObject = new();
+    private static readonly List<PollConsul> ServiceDiscoveryProviders = new(); // TODO It must be scoped service in DI-container
```

Comment:
Agree, it's not good at the minute.

Reply:
Indeed, it is necessary. However, there is no time to refactor further because it's beyond the scope of the current bug fix. For now, the comments in src are noted for future refactoring. The present solution may not be elegant, but it functions. I have prepared a [TODO task](https://github.com/ThreeMammals/Ocelot/pull/2151#issuecomment-2387973313) for the next phase of refactoring, which has been added to https://github.com/ThreeMammals/Ocelot/issues/2141#issuecomment-2283697861 (item 5).